publish/tests/test_version.py

```diff
@@ -1,8 +1,25 @@
 import pytest
 
+from publish.models import Version
 from .fuzzy import DandisetIdentifierRe, TimestampRe, VersionRe
 
 
+@pytest.mark.django_db
+def test_version_make_version_nosave(dandiset, version_factory):
+    # Without saving, the output should be reproducable
+    version_str_1 = Version.make_version(dandiset)
+    version_str_2 = Version.make_version(dandiset)
+    assert version_str_1 == version_str_2
+
+
+@pytest.mark.django_db
+def test_version_make_version_save(dandiset, version_factory):
+    # Given an existing version at the current time, a different one should be allocated
+    version_1 = version_factory(dandiset=dandiset)
```

Comment:
Have you verified that `version_factory` will actually invoke `make_version` when initializing the new object? I can imagine this test having false negatives if it just guesses a valid-looking version string.

Reply:
It's a good point. I can add a Spy to ensure it's called.